pkg/controller/monitor/cnr_indicator.go

```diff
@@ -0,0 +1,207 @@
+				}
+				if qos.IsPodNumaBinding(qosConf, pod) && qos.IsPodNumaExclusive(qosConf, pod) && podNum > 1 {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+// checkNumaAllocatableSumAnomaly checks whether the node's sum of numa allocatable is not equal to the node allocatable
+func (ctrl *CNRMonitorController) checkNumaAllocatableSumAnomaly(cnr *v1alpha1.CustomNodeResource) bool {
+	// refer to https://github.com/kubewharf/katalyst-core/blob/main/pkg/agent/resourcemanager/fetcher/system/systemplugin.go#L126
+	// If the order of cpu and memory changes, the sequence number needs to be changed here as well
+	if len(cnr.Spec.NodeResourceProperties) < 3 {
+		klog.Errorf("[CNRIndicatorNumaAllocatableSumAnomaly] failed to get nodeResourceProperties")
+		return false
+	}
+	nodeCpuAllocatable, nodeMemAllocatable := int(cnr.Spec.NodeResourceProperties[1].PropertyQuantity.AsApproximateFloat64()), int(cnr.Spec.NodeResourceProperties[2].PropertyQuantity.AsApproximateFloat64())
+	klog.Infof("[CNRIndicatorNumaAllocatableSumAnomaly] nodeCpuAllocatable: %d, nodeMemAllocatable: %d", nodeCpuAllocatable, nodeMemAllocatable)
+	numaCpuAllocatableSum, numaMemAllocatableSum := 0, 0
+	for _, socket := range cnr.Status.TopologyZone {
+		for _, numa := range socket.Children {
+			if numa.Type != v1alpha1.TopologyTypeNuma {
+				// only check numa
+				continue
+			}
+			numaCpuAllocatableSum += int(numa.Resources.Allocatable.Cpu().AsApproximateFloat64())
+			numaMemAllocatableSum += int(numa.Resources.Allocatable.Memory().AsApproximateFloat64())
+		}
+	}
+	klog.Infof("[CNRIndicatorNumaAllocatableSumAnomaly] numaCpuAllocatableSum: %d, numaMemAllocatableSum: %d", numaCpuAllocatableSum, numaMemAllocatableSum)
+	if numaCpuAllocatableSum != nodeCpuAllocatable || numaMemAllocatableSum != nodeMemAllocatable {
+		return true
+	}
+	return false
+}
+
+// checkPodAllocationSumAnomaly checks whether the numa's sum of pod allocation is greater than the numa allocatable
+func (ctrl *CNRMonitorController) checkPodAllocationSumAnomaly(cnr *v1alpha1.CustomNodeResource) bool {
+	qosConf := generic.NewQoSConfiguration()
+	for _, socket := range cnr.Status.TopologyZone {
+		for _, numa := range socket.Children {
+			if numa.Type != v1alpha1.TopologyTypeNuma {
+				// only check numa
+				continue
+			}
+			numaCpuAllocatable, numaMemAllocatable := int(numa.Resources.Allocatable.Cpu().AsApproximateFloat64()), int(numa.Resources.Allocatable.Memory().AsApproximateFloat64())
+			klog.Infof("[CNRIndicatorPodAllocationSumAnomaly] numaCpuAllocatable: %d, numaMemAllocatable: %d", numaCpuAllocatable, numaMemAllocatable)
+			podCpuAllocationSum, podMemAllocationSum := 0, 0
+			for _, allocation := range numa.Allocations {
+				key := allocation.Consumer
+				namespace, podname, _, err := native.ParseUniqObjectUIDKey(key)
+				if err != nil {
+					klog.Errorf("[CNRIndicatorPodAllocationSumAnomaly] failed to parse uniq object uid key %s", key)
```

Comment:
If err is not nil, continue to avoid unexpected returns that could lead to panic in subsequent logic. This rule should be followed for all error checks in the code.

Reply:
Done, thanks